common.go

```diff
@@ -141,12 +141,58 @@ func computePayloadAuthenticator(macKey macKey, payloadHash payloadHash) payload
 	return sliceToByte32(fullMAC[:cryptoAuthBytes])
 }
 
-func computeMACKey(secret BoxSecretKey, public BoxPublicKey, headerHash headerHash) macKey {
-	nonce := nonceForMACKeyBox(headerHash)
+func computeMACKeySingle(secret BoxSecretKey, public BoxPublicKey, nonce *Nonce) macKey {
 	macKeyBox := secret.Box(public, nonce, make([]byte, cryptoAuthKeyBytes))
 	return sliceToByte32(macKeyBox[poly1305.TagSize : poly1305.TagSize+cryptoAuthKeyBytes])
 }
 
+func sum512Truncate256(in []byte) [32]byte {
+	// Consistent with computePayloadAuthenticator in that it
+	// truncates SHA512 instead of calling SHA512/256, which has
+	// different IVs.
+	sum512 := sha512.Sum512(in)
+	return sliceToByte32(sum512[:32])
+}
+
+func computeMACKeySender(version Version, index uint64, secret, eSecret BoxSecretKey, public BoxPublicKey, headerHash headerHash) macKey {
+	switch version {
+	case Version1():
+		nonce := nonceForMACKeyBoxV1(headerHash)
+		return computeMACKeySingle(secret, public, nonce)
+	case Version2():
+		nonce := nonceForMACKeyBoxV2(headerHash, index)
+		mac1 := computeMACKeySingle(secret, public, nonce)
+		mac2 := computeMACKeySingle(eSecret, public, nonce)
```

Comment:
Would it be good practice to use a different nonce space for `mac2`? The `secret` and `eSecret` keys _are_ the same, when you're in anonymous sender mode, so this counts as nonce reuse. It does seem safe in this case, but then again...those are famous last words :-D